rpc/legacyrpc/methods.go

```diff
@@ -3442,6 +3444,107 @@ WrongAddrKind:
 	return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "address must be secp256k1 P2PK or P2PKH")
 }
 
+func deriveCoinTypeKey(seed []byte, coinType uint32, params *chaincfg.Params) (*hdkeychain.ExtendedKey, error) {
+	// Create new root from the inputted seed and the current net
+	root, err := hdkeychain.NewMaster(seed[:], params)
+	if err != nil {
+		return nil, err
+	}
+
+	// BIP0032 hierarchy: m/<purpose>'/
+	// Where purpose = 44 and the ' indicates hardening with the HardenedKeyStart 0x80000000
+	purpose, err := root.Child(44 + hdkeychain.HardenedKeyStart)
+	if err != nil {
+		return nil, err
+	}
+	defer purpose.Zero()
+
+	// BIP0044 hierarchy: m/44'/<coin type>'
+	// Where coin type is either the legacy coin type, 20, or the coin type described in SLIP0044, 44.
+	coinTypePrivKey, err := purpose.Child(coinType + hdkeychain.HardenedKeyStart)
+	if err != nil {
+		return nil, err
+	}
+	defer coinTypePrivKey.Zero()
+
+	return coinTypePrivKey, nil
+}
+
+// verifySeed checks if a user inputted seed is that of the wallet by comparing their child key derivatied
+// public keys. It returns a bool if this the case as well as the current coin type of the wallet.  An optional
+// parameter is also avalaible that checks beyond default accounts. It returns the result using the RPC api.
+func verifySeed(s *Server, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*dcrjson.VerifySeedCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+
+	coinType, err := w.CoinType()
+	if err != nil {
+		return nil, err
+	}
+
+	decodedSeed, err := walletseed.DecodeUserInput(cmd.Seed)
+	if err != nil {
+		return nil, err
+	}
+
+	// Changed inputted seed, type string, to type byte[] so hdkeychain methods can be utilize using DecodeUserInput
+	// and run then derivedCoinTypeKey to receive the coin type private key.
+	coinTypePrivKey, err := deriveCoinTypeKey(decodedSeed, coinType, w.ChainParams())
+	if err != nil {
+		return nil, err
+	}
+	defer coinTypePrivKey.Zero()
+
+	// Grab coin type private key from wallet for future comparison to the derived inputted coin type private key
+	walletCoinTypePrivKey, err := w.CoinTypeKey()
+	if err != nil {
+		return nil, err
+	}
+	defer walletCoinTypePrivKey.Zero()
+
+	var matches bool
+	switch {
+	case cmd.Account != nil:
```

Comment:
An account variable should be created and defaulted to zero.  When cmd.Account is non-nil, the default can be changed, and then the rest of the function can continue.  This can happen near the beginning of the method, and then rest of the method can just operate on the account variable.